v2/component_analyses.yaml

```diff
@@ -0,0 +1,288 @@
+openapi: 3.0.1
+info:
+  title: Analytics Core API Specs for Version 2
+  version: "2"
+servers:
+- url: https://f8a-analytics-2445582058137.production.gw.apicast.io/api/v2
+  description: "Fabric8 Server"
+security:
+- tokenAuth: []
+paths:
+  /component-analyses/{eco}/{name}/{ver}:
+    get:
+      tags:
+      - Scan Services
+      summary: Get security and other details about a software component and recommendations
+        to change to a more secured version
+      operationId: f8a_scanner.api_v2.component-analyses
+      parameters:
+      - name: eco
+        in: path
+        description: component ecosystem
+        required: true
+        schema:
+          type: string
+      - name: name
+        in: path
+        description: component name
+        required: true
+        schema:
+          type: string
+      - name: ver
+        in: path
+        description: component version
+        required: true
+        schema:
+          type: string
+      - name: security-vendor
+        in: header
+        description: Security Vendor
+        schema:
+          type: string
+          enum:
+          - snyk
+      responses:
+        200:
+          description: Detailed response for the requested application component
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ComponentAnalysesResponse'
+        202:
+          description: Package is currently Unavailable. Please try again later
+          content: {}
+        401:
+          description: Request unauthorized
+          content: {}
+        404:
+          description: Data not found
+          content: {}
+components:
+  schemas:
+    Version:
+      title: Component Version Details
+      type: object
+      properties:
+        pecosystem:
+          type: array
+          items:
+            type: string
+        pname:
+          type: array
+          items:
+            type: string
+        version:
+          type: array
+          items:
+            type: string
+        declared_licenses:
+          type: array
+          items:
+            type: string
+        licenses:
+          type: array
+          items:
+            type: string
+        last_updated:
+          type: array
+          items:
+            type: number
+            format: float
+        vulnerability_ids:
```

Comment:
How do flag a vulnerability as public / private ?

Reply:
Added different Models each for Private and Public Vulnerabilities.
So, if user is Registered,  API response will have `private_vulnerabilities:` and `exploitable_vulnerabilities` in response.